packages/hardhat/contracts/GardensTemplate.sol

```diff
@@ -19,72 +26,123 @@ contract GardensTemplate is BaseTemplate, AppIdsXDai {
+
     mapping(address => DeployedContracts) internal senderDeployedContracts;
+    IHoneyswapRouter public honeyswapRouter;
+    ERC20 public honeyToken;
+    address public stableToken;
+    IPriceOracle public honeyPriceOracle;
+    IIncentivisedPriceOracleFactory public incentivisedPriceOracleFactory;
+    ICollateralRequirementUpdaterFactory public collateralRequirementUpdaterFactory;
+    IUniswapV2Factory public uniswapFactory;
+    address public arbitrator;
+    address public stakingFactory;
 
     constructor(
         DAOFactory _daoFactory,
         ENS _ens,
         MiniMeTokenFactory _miniMeFactory,
-        IFIFSResolvingRegistrar _aragonID
+        IFIFSResolvingRegistrar _aragonID,
+        IHoneyswapRouter _honeyswapRouter,
+        ERC20 _honeyToken,
+        address _stableToken,
+        IPriceOracle _honeyPriceOracle,
+        IIncentivisedPriceOracleFactory _incentivisedPriceOracleFactory,
+        ICollateralRequirementUpdaterFactory _collateralRequirementUpdaterFactory,
+        IUniswapV2Factory _uniswapFactory,
+        address _arbitrator,
+        address _stakingFactory
     ) public BaseTemplate(_daoFactory, _ens, _miniMeFactory, _aragonID) {
         _ensureAragonIdIsValid(_aragonID);
         _ensureMiniMeFactoryIsValid(_miniMeFactory);
+        honeyswapRouter = _honeyswapRouter;
+        honeyToken = _honeyToken;
+        stableToken = _stableToken;
+        honeyPriceOracle = _honeyPriceOracle;
+        incentivisedPriceOracleFactory = _incentivisedPriceOracleFactory;
+        collateralRequirementUpdaterFactory = _collateralRequirementUpdaterFactory;
+        uniswapFactory = _uniswapFactory;
+        arbitrator = _arbitrator;
+        stakingFactory = _stakingFactory;
     }
 
     // New DAO functions //
 
     /**
      * @dev Create the DAO and initialise the basic apps necessary for gardens
-     * @param _voteTokenName DAO governance new token name
-     * @param _voteTokenSymbol DAO governance new token symbol
-     * @param _fundingPoolStake Initial amount of tokens in the funding pool
+     * @param _existingToken An existing token used for the common pool token. Set to address(0) to create a new token.
+     * @param _gardenTokenName DAO governance new token name
+     * @param _gardenTokenSymbol DAO governance new token symbol
+     * @param _commonPoolAmount Initial amount of tokens in the funding pool. Unused if _existingToken is set.
+     * @param _gardenTokenLiquidity Amount of tokens to use as liquidity, paired with Honey. Unused if _existingToken is set.
      * @param _disputableVotingSettings Array of [voteDuration, voteSupportRequired, voteMinAcceptanceQuorum, voteDelegatedVotingPeriod,
      *    voteQuietEndingPeriod, voteQuietEndingExtension, voteExecutionDelay] to set up the voting app of the organization
      */
     function createDaoTxOne(
-        string _voteTokenName,
-        string _voteTokenSymbol,
-        uint256 _fundingPoolStake,
+        ERC20 _existingToken,
```

Comment:
I have considered it but unless I need to do it to prevent stack too deep errors I doubt I will because it would require a bunch of internal functions that reference stuff created in this function. It would look pretty ugly and be harder to follow.

Reply:
I have considered it. Will try and re-arrange.

Reply:
Decided against it in the end. Would require more security checks and I don't think it's hard to read and understand as it is. If it was then it would be worth splitting it up.